### PR TITLE
Add ARC to Capstone Printer

### PR DESCRIPTION
**Pull request by R33v0LT**
This PR adds ARC support to PrinterCapstone for successful generation of tablegen files

### CODE REVIEW

**Comment by Rot127** [comment]
I keep it open until the Capstone PR is done. Just in case you need to do some fixes.

File: llvm/utils/TableGen/PrinterCapstone.cpp [hunk 1]
```diff
@@ -3069,6 +3069,10 @@ std::string getCSOperandType(
     std::map<std::string, std::vector<Record *>> const InsnPatternMap) {
   std::string OperandType = getPrimaryCSOperandType(OpRec);
 
+  // $b3 in ARC is considered CS_OP_INVALID. Forcing it to be CS_OP_REG
+  if ((StringRef(TargetName).upper() == "ARC") && (OpName.str() == "b3")) {
```
Comment (Rot127) [inline llvm/utils/TableGen/PrinterCapstone.cpp]: You can check if the `let OperandType = OPERAND_REGISTER` is set (in the `td` files). If not, this is why you likely get the `CS_OP_INVALID`.

Comment (Rot127) [reply llvm/utils/TableGen/PrinterCapstone.cpp]: This is ok though. But only if it is stays a simple case.
Please move this `if` condition into a `static inline` function with a nice name though.

Comment (R33v0LT) [reply llvm/utils/TableGen/PrinterCapstone.cpp]: b3 belongs to the GPR32Reduced class, which does not have an OperandType. Added it

**Pull request merged**